### PR TITLE
[libc] Remove recursion in get_object_files_for_test to improve build time.

### PR DESCRIPTION
**Pull request by lntue**

### CODE REVIEW

File: libc/cmake/modules/LLVMLibCTestRules.cmake [hunk 1]
```diff
@@ -15,48 +15,60 @@
...
+        # Skip any tests whose dependencies have not been defined.
         list(APPEND skipped_list ${dep})
         continue()
       endif()
-      get_target_property(object_file_raw ${dep} "OBJECT_FILE_RAW")
-      if(object_file_raw)
-        list(APPEND object_files ${object_file_raw})
+      get_target_property(dep_type ${dep} "TARGET_TYPE")
+      if(NOT dep_type)
+        # Target for which TARGET_TYPE property is not set do not
```
Comment (gchatelet) [inline libc/cmake/modules/LLVMLibCTestRules.cmake:34]: [nit] Skip tests with no object dependencies.

File: libc/cmake/modules/LLVMLibCTestRules.cmake [hunk 2]
```diff
@@ -15,48 +15,60 @@
...
+        if(is_skipped)
+          list(APPEND skipped_list ${dep})
+          continue()
+        endif()
+        get_target_property(object_file_raw ${dep} "OBJECT_FILE_RAW")
+        if(object_file_raw)
+          list(APPEND object_files ${object_file_raw})
+        endif()
+      elseif(${dep_type} STREQUAL ${ENTRYPOINT_OBJ_VENDOR_TARGET_TYPE})
+        # We skip tests for all externally implemented entrypoints.
```
Comment (gchatelet) [inline libc/cmake/modules/LLVMLibCTestRules.cmake:55]: [nit] Skip tests for externally implemented entrypoints.

**Review by gchatelet — APPROVED** [review]
LGTM
Only a few nits to get comments short and consistent,

File: libc/cmake/modules/LLVMLibCTestRules.cmake [hunk 3]
```diff
@@ -15,48 +15,60 @@
...
-      endif()
-    elseif(${dep_type} STREQUAL ${ENTRYPOINT_OBJ_TARGET_TYPE})
-      get_target_property(is_skipped ${dep} "SKIPPED")
-      if(is_skipped)
+  while(length)
+    set(indirect_list "")
+
+    foreach(dep IN LISTS unchecked_list)
+      if (NOT TARGET ${dep})
+        # Skip any tests whose dependencies have not been defined.
```
[the remark this answers is not in the document]
Comment (lntue) [reply libc/cmake/modules/LLVMLibCTestRules.cmake]: Done.

File: libc/cmake/modules/LLVMLibCTestRules.cmake (hunk 1 above)
Comment (lntue) [reply libc/cmake/modules/LLVMLibCTestRules.cmake]: Done.

File: libc/cmake/modules/LLVMLibCTestRules.cmake (hunk 2 above)
Comment (lntue) [reply libc/cmake/modules/LLVMLibCTestRules.cmake]: Done.

**Pull request merged**